src/components/plot/template-params.tsx

```diff
@@ -0,0 +1,47 @@
+import { createContext, useState, ReactNode } from 'react';
+import { PlotTemplateParams } from '@isrd-isi-edu/deriva-webapps/src/hooks/chart';
+
+type TemplateParamsProps = {
+  children: ReactNode;
+}
+
+type ITemplateContext = {templateParams: PlotTemplateParams, setTemplateParams: React.Dispatch<React.SetStateAction<PlotTemplateParams>>,
+  selectorOptionChanged: boolean, setSelectorOptionChanged: React.Dispatch<React.SetStateAction<boolean>> };
+
+// Define the context with the correct initial value
+
+export const TemplateParamsContext = createContext<ITemplateContext>({templateParams: { 
```

Comment:
Move this to a file in `hooks` similar to what was done in chaise for other context/providers.